proposals/0000-down_ranked_exists.rst

```diff
@@ -0,0 +1,462 @@
+
+Currently, every existential must be encoded using its own datatype, which is laborious. Furthermore, packing and unpacking these datatypes must be done by hand, which is cluttersome.
+
+UpRanked Existentials
+~~~~~~~~~~~~~~~~~~~~~
+
+Alternative Proposal is "First-class existential types" `#473`_ .
+
+#473 suggests to add *de facto* **UpRanked Existential**
+
+- **UpRanked Existential** rule: Any N-Ranked ``forall`` *type_variable* is INSTEAD (N+1)-Ranked ``exists`` *type_variable* 
+::
+
+  --UpRanked Existentials:
+  cmpg :: forall a. a -> a ->(forall b. b -> b -> Bool) -> Bool
+  cmpg x y f = f x y
+
+  -- same as
+  cmpg :: forall a. a -> a ->((exists b. b -> b) -> Bool) -> Bool
+
+  -- same as
+  cmpg :: (exists a. a -> a ->((exists b. b -> b) -> Bool)) -> Bool
+
+Unfortunately, working with High-Ranked Types is complicated in Haskell and it's Core.
+
+DownRanked Existentials
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This Proposal suggest to add the opposite (of #473) : **DownRanked Existential**
+
+- **DownRanked Existential** rule: Any N-Ranked ``forall`` (or ``exists``) *type_variable* is ALSO (N-1)-Ranked ``exists`` *type_variable* 
+::
+
+  --DownRanked Existentials:
+  cmpg2 :: forall a. a -> a ->(forall b. b -> b -> (b, Bool)) -> (a, Bool)
+  cmpg2 x y f = f x y
+
+  -- same as
+  cmpg2 :: forall a. exists b. a -> a ->(forall b. b -> b -> (b, Bool)) -> (b, Bool)
+
+
+The Main rule: *if* ∀a: f a *then* ∃b: ∀a, a ≡ b : f b
+
+Second rule: *if* ∃a: f a *then* ∃b: ∃a, a ≡ b : f b
+
+This is the core idea of this Dependent existential type from Higher-Ranked (Exactly N+1 Ranked) ``forall`` / ``exists`` type variables with `a ≡ b` equality condition.
+
+*Note: using same keyword "exists" for both UpRanked and DownRanked Existential Quantifiers is incompatible and inconsistent idea*
+
+
+Proposed Change Specification
+-----------------------------
+
+Main and second rules give us unique proprieties of DownRanked existentials.
+
+Roles
+~~~~~
+
+1. Boxing type variable into a Data declaration 
+::
+
+  -- hidden conventional existential GHC type
+  data Box = forall a. MkBox a
+
+  -- open existential type
+  data exists a. Ex = forall a. MkEx { unEx :: a }  -- NEW!
+
+Main and second rules say that in open existential type we capture on N-Rank **same** *type variable* which escaped from (N+1 Ranked) ``forall`` (or ``exists`` )
+
+2. Extractor / unboxing / escaping from Data-constructor 
+::
+
+  fromBox :: Box -> ???
+  fromBox (MkBox x) = x       -- Error!
+
+  fromEx :: exists a. Ex -> a
+  fromEx (MkEx x) = x         -- OK! NEW!
+
+  fromEx2 :: exists a. Ex -> a
+  fromEx2 = unEx              -- OK! NEW!
+
+Main and second rules guarantee us that unboxing give us same type as boxing or it is bottom type (for phantom or partly phantom existentials)
+
+3. Direct capture type variable by Data-constructor
+::
+
+  toBox :: forall a. a -> Box
+  toBox = MkBox
+
+  toEx :: forall a. a -> exists a. Ex
```

Comment:
Again, why shadowing `a` and quantifying with `exists a` when `Ex` contains no free `a`?